gvm/protocols/gmpv208/entities/hosts.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2021 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# pylint:  disable=redefined-builtin
+
+from typing import Any, Optional
+
+from gvm.errors import RequiredArgument
+from gvm.utils import add_filter
+from gvm.xml import XmlCommand
+
+
+class HostsMixin:
+    def create_host(self, name: str, *, comment: Optional[str] = None) -> Any:
+        """Create a new host host
+
+        Arguments:
+            name: Name for the new host host
+            comment: Comment for the new host host
+
+        Returns:
+            The response. See :py:meth:`send_command` for details.
+        """
+        if not name:
+            raise RequiredArgument(
+                function=self.create_host.__name__, argument='name'
+            )
+
+        cmd = XmlCommand("create_asset")
+        host = cmd.add_element("asset")
+        host.add_element("type", "host")  # ignored for gmp7, required for gmp8
+        host.add_element("name", name)
+
+        if comment:
+            host.add_element("comment", comment)
+
+        return self._send_xml_command(cmd)
+
+    def delete_host(
+        self, *, host_id: Optional[str] = None, report_id: Optional[str] = None
```

Comment:
This api looks very bad. both arguments are optional but in reality one of them is required. I would either split deleting a host by report_id into another function or just remove that functionality.

Reply:
yeah probably a good idea. I didn't look into all the functions in detail.